### DISCUSSION TITLE
Handling PointGrey timestamps with Spinnaker SDK

### DISCUSSION THREAD

**Discussion by NidhiGujar** [bonsai-rx/bonsai#1672] [Q&A] [answered]
Hello! I am using Bonsai for my animal behavior experiments. I am using a PointGrey camera with Spinnaker SDK. 
I am facing trouble decoding the timestamps given out by Bonsai using a Python code.
Any help would be truly appreciated.

**Comment by bruno-f-cruz** [comment]
Can you add some code (bonsai and Python) and snippets of data?

**Reply by NidhiGujar** [reply]
Apologies for the delay. Following are the codes I've been using - 
 
![Bonsai_workflow](https://github.com/bonsai-rx/bonsai/assets/101789311/7bcc764e-40f9-4b5c-a6e5-799e3af8f8ed)
This is my Bonsai workflow.

![Python_code](https://github.com/bonsai-rx/bonsai/assets/101789311/4b2714a9-6837-4654-a3fa-5e97949ce587)
These are the functions that I am trying to use.

[tstamp_5.csv](https://github.com/bonsai-rx/bonsai/files/14291611/tstamp_5.csv)
And this is an example of a timestamp file for a given trial. The duration of this trial was 10 secs.

**Reply by bruno-f-cruz** [reply, marked as answer]
Not sure I follow why you need that convert function at all. The values outputted by the Spinnaker SDK are already in nanoseconds. Cant you just use the csv python library and read the raw values? If i am interpreting correctly it looks like your camera is running at 150fps.

**Reply by NidhiGujar** [reply]
I guess this solves it. I did not know that the timestamp values given out by Spinnaker are already in nanoseconds. 
Thanks for the prompt reply!
